common/src/main/java/com/brandwatch/minibcr/common/repository/QueryRepositoryImpl.java

```diff
@@ -0,0 +1,69 @@
+package com.brandwatch.minibcr.common.repository;
+
+import java.util.List;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.dao.EmptyResultDataAccessException;
+import org.springframework.jdbc.core.namedparam.MapSqlParameterSource;
+import org.springframework.jdbc.core.namedparam.NamedParameterJdbcTemplate;
+import org.springframework.stereotype.Repository;
+
+import com.brandwatch.minibcr.common.domain.Query;
+
+@Repository
+public class QueryRepositoryImpl implements QueryRepository {
+
+    private static final String QUERY_TABLE = "queries";
+    private static final String ALL_FIELDS = "id, definition";
+    private final Logger logger = LoggerFactory.getLogger(this.getClass());
+    private final QueryRowMapper queryRowMapper;
+    private final NamedParameterJdbcTemplate namedParameterJdbcTemplate;
```

Comment:
I'd probably name this something a little shorter 

```suggestion
    private final NamedParameterJdbcTemplate jdbcTemplate;
```

Reply:
So I avoided this shorter name in particular because there is another template called `jdbcTemplate` which `NamedParameterJdbcTemplate` extends. You can also check in the same class in deleteAll method where I use `jdbcTemplate` on its own.